sources/imdsclient/src/lib.rs

```diff
@@ -351,23 +354,39 @@ fn build_public_key_targets(public_key_list: &str) -> Vec<String> {
 /// Helper to fetch an IMDSv2 session token that is valid for 60 seconds.
 async fn fetch_token(client: &Client, imds_base_uri: &str) -> Result<String> {
     let uri = format!("{}/{}", imds_base_uri, SESSION_TARGET);
-    let response = client
-        .put(&uri)
-        .header("X-aws-ec2-metadata-token-ttl-seconds", "60")
-        .send()
-        .await
-        .context(error::Request {
-            method: "PUT",
-            uri: &uri,
-        })?
-        .error_for_status()
-        .context(error::BadResponse { uri: &uri })?;
-    let code = response.status();
-    response.text().await.context(error::ResponseBody {
-        method: "PUT",
-        uri,
-        code,
-    })
+    let mut token_attempts: u8 = 0;
+    let max_token_attempts: u8 = 2;
+    loop {
+        token_attempts += 1;
+        ensure!(
+            token_attempts <= max_token_attempts,
+            error::FailedFetchToken { token_attempts }
+        );
+        if token_attempts > 1 {
+            time::sleep(Duration::from_secs(5)).await;
+        }
```

Comment:
I'd like to align the number of retries and the length of the delay with `fetch_imds`  - the requests are going to the same service and reaching the failure state in either function will likely result in a boot failure.

Reply:
The retries in `fetch_imds` are mostly to trigger token refreshes in the even the token is expired as opposed to IMDS unavailability (hence a shorter sleep). Each client initialization and `fetch_imds` attempt is going to fetch a new token. If IMDS is unreachable for whatever reason, the failure would occur during the `fetch_token`. Since we are adding retries to the token fetch now it might make sense to drop the max_imds_attempts from 3 to 2, but I wouldn't recommend aligning the sleep duration.

Reply:
Instead of dropping `fetch_imds` from 3 to 2, I bumped `fetch_token` from 2 to 3.